### PR TITLE
[SD-448] User Events

### PR DESCRIPTION
**Pull request by rjungemann**

### CODE REVIEW

File: helpers/userEvents.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const streamName = process.env.USER_ANALYTICS_STREAM_NAME;
+    const env = process.env.NODE_ENV;
+    console.log('options', endpoint, token);
+    const userAnalytics = new UserAnalytics({
+        token: token,
+        endpoint: endpoint,
+    });
+    const mobileDetect = new MobileDetect(req.headers['user-agent']);
+    const data = {
+        distinct_id: '',
```
Comment (rjungemann) [inline helpers/userEvents.js]: Should this possibly be `user.tracking_id`? I'm fairly certain that `user.id` would not be kosher here for GDPR reasons. Thoughts?

Can `user.tracking_id` be accessed from anywhere else? It isn't in the response of the `getAccounts` JSON-RPC method.

File: helpers/userEvents.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+var MobileDetect = require('mobile-detect');
+var UserAnalytics = require('user-analytics');
```
Comment (bnchdrff) [inline helpers/userEvents.js]: `const` would be good for these two guys

File: package.json [hunk 3]
```diff
@@ -30,12 +30,15 @@
     "isomorphic-fetch": "2.2.1",
     "jsonwebtoken": "8.1.1",
     "maxmind": "2.4.0",
+    "mobile-detect": "^1.4.3",
```
[the remark this answers is not in the document]
Comment (bnchdrff) [reply package.json]: also pin to an exact release

File: helpers/userEvents.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+function UserEvents() {}
+
+UserEvents.trackSignup = function(req, user) {
+    const endpoint = process.env.USER_ANALYTICS_ENDPOINT;
+    const token = process.env.USER_ANALYTICS_TOKEN;
+    const streamName = process.env.USER_ANALYTICS_STREAM_NAME;
+    const env = process.env.NODE_ENV;
+    console.log('options', endpoint, token);
```
Comment (bnchdrff) [inline helpers/userEvents.js]: looks like we can ditch this

File: helpers/userEvents.js (hunk 1 above)
Comment (bnchdrff) [reply helpers/userEvents.js]: a user of faucet won't have an on-chain account at this point -- if we want to use some sort of id here we can use our `X-Request-Id` (see app.js)

File: helpers/userEvents.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const userAnalytics = new UserAnalytics({
+        token: token,
+        endpoint: endpoint,
+    });
```
Comment (bnchdrff) [inline helpers/userEvents.js]: this setup work can be done once during instantiation of this module, rather than each time it's called

**Pull request closed without merging**